apps/emqx_bridge_disk_log/src/emqx_bridge_disk_log_connector.erl

```diff
@@ -0,0 +1,438 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+-module(emqx_bridge_disk_log_connector).
+
+-feature(maybe_expr, enable).
+
+-behaviour(emqx_resource).
+
+-include_lib("kernel/include/file.hrl").
+-include_lib("emqx/include/logger.hrl").
+-include_lib("snabbkaffe/include/trace.hrl").
+-include_lib("emqx_resource/include/emqx_resource.hrl").
+-include_lib("emqx/include/emqx_trace.hrl").
+-include("emqx_bridge_disk_log.hrl").
+
+%% `emqx_resource' API
+-export([
+    resource_type/0,
+    callback_mode/0,
+
+    on_start/2,
+    on_stop/2,
+    on_get_status/2,
+
+    on_get_channels/1,
+    on_add_channel/4,
+    on_remove_channel/3,
+    on_get_channel_status/3,
+
+    on_query/3,
+    on_batch_query/3
+]).
+
+-ifdef(TEST).
+-export([flush/1, get_wrap_logs/2]).
+-endif.
+
+-elvis([{elvis_style, export_used_types, disable}]).
+
+%%------------------------------------------------------------------------------
+%% Type declarations
+%%------------------------------------------------------------------------------
+
+%% Allocatable resources
+-define(disk_log, disk_log).
+
+-define(filepath, filepath).
+-define(installed_actions, installed_actions).
+
+-define(template, template).
+-define(write_mode, write_mode).
+
+-type connector_config() :: #{
+    filepath := binary(),
+    max_file_size := pos_integer(),
+    max_file_number := pos_integer()
+}.
+-type connector_state() :: #{
+    ?filepath := binary(),
+    ?installed_actions := #{action_resource_id() => action_state()}
+}.
+
+-type action_config() :: #{
+    parameters := #{
+        template := binary(),
+        write_mode := write_mode()
+    }
+}.
+-type action_state() :: #{
+    ?template := emqx_template:t(),
+    ?write_mode := write_mode()
+}.
+
+-type query() :: {_Tag :: channel_id(), _Data :: map()}.
+
+-type log_name() :: connector_resource_id().
+-type write_mode() :: sync | async.
+
+%%------------------------------------------------------------------------------
+%% `emqx_resource' API
+%%------------------------------------------------------------------------------
+
+-spec resource_type() -> atom().
+resource_type() ->
+    disk_log.
+
+-spec callback_mode() -> callback_mode().
+callback_mode() ->
+    always_sync.
+
+-spec on_start(connector_resource_id(), connector_config()) ->
+    {ok, connector_state()} | {error, _Reason}.
+on_start(ConnResId, ConnConfig) ->
+    #{filepath := FilepathBin} = ConnConfig,
+    maybe
```

Comment:
Nit: this is basically a case 😅

Reply:
Yes, but I see no harm in using `maybe` in such occasions.  :smile_cat: